perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet_node.hpp

```diff
@@ -0,0 +1,221 @@
+
+#include "bevdet.hpp"
+#include "cpu_jpegdecoder.hpp"
+#include "tf2_ros/buffer.h"
+#include "tf2_ros/transform_listener.h"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <opencv2/opencv.hpp>
+
+#include <autoware_perception_msgs/msg/detected_object_kinematics.hpp>
+#include <autoware_perception_msgs/msg/detected_objects.hpp>
+#include <autoware_perception_msgs/msg/object_classification.hpp>
+#include <autoware_perception_msgs/msg/shape.hpp>
+#include <geometry_msgs/msg/point32.hpp>
+#include <geometry_msgs/msg/quaternion.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+
+#include <cv_bridge/cv_bridge.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/synchronizer.h>
+#include <pcl/common/transforms.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>  // msg2pcl
+#include <yaml-cpp/yaml.h>
+
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+uint8_t getSemanticType(const std::string & class_name);
+
+void box3DToDetectedObjects(
+  const std::vector<Box> & boxes, autoware_perception_msgs::msg::DetectedObjects & objects,
+  const std::vector<std::string> & class_names, float score_thre, const bool has_twist);
+
+// Get the rotation and translation from a geometry_msgs::msg::TransformStamped
+void getTransform(
+  const geometry_msgs::msg::TransformStamped & transform, Eigen::Quaternion<float> & rot,
+  Eigen::Translation3f & translation);
+
+// Get the camera intrinsics from a sensor_msgs::msg::CameraInfo
+void getCameraIntrinsics(
+  const sensor_msgs::msg::CameraInfo::SharedPtr msg, Eigen::Matrix3f & intrinsics);
+
+class TRTBEVDetNode : public rclcpp::Node
+{
+  /**
+   * @class TRTBEVDetNode
+   * @brief This class represents a node for performing object detection using TensorRT on BEV
+   * (Bird's Eye View) images.
+   */
+private:
+  size_t img_N_;    ///< Number of images
+  uint32_t img_w_;  ///< Width of the images
+  uint32_t img_h_;  ///< Height of the images
+
+  std::string model_config_;  ///< Path to the model configuration file
+
+  std::string onnx_file_;    ///< Path to the ONNX file
+  std::string engine_file_;  ///< Path to the TensorRT engine file
+
+  std::vector<std::string> imgs_name_;    ///< Names of the images
+  std::vector<std::string> class_names_;  ///< Names of the object classes
+
+  camsData sampleData_;  ///< Sample data for camera parameters
+  std::shared_ptr<autoware::tensorrt_bevdet::BEVDet>
+    bevdet_;  ///< Object for performing object detection
+
+  uchar * imgs_dev_ = nullptr;  ///< Device pointer for storing the images
```

Comment:
The comment `///< Device pointer for storing the images` for `imgs_dev_` is good. However, consider adding a comment about memory management;  how and when is this memory allocated and freed?